mcgill-augmented-reality-backend/src/main/java/mcgillar/backend/controllers/user/AppUserInfoController.java

```diff
@@ -0,0 +1,108 @@
+package mcgillar.backend.controllers.user;
+
+import org.springframework.http.HttpStatus;
+import org.springframework.http.ResponseEntity;
+import org.springframework.security.core.Authentication;
+import org.springframework.web.bind.annotation.PostMapping;
+import org.springframework.web.bind.annotation.RequestMapping;
+import org.springframework.web.bind.annotation.RequestParam;
+import org.springframework.web.bind.annotation.RestController;
+
+import lombok.AllArgsConstructor;
+import mcgillar.backend.TO.user.UserInfoTO;
+import mcgillar.backend.model.user.AppUser;
+import mcgillar.backend.model.user.AppUserInfo;
+import mcgillar.backend.services.user.AppUserInfoService;
+import mcgillar.backend.services.user.AppUserService;
+
+@AllArgsConstructor
```

Comment:
Do we need this tag? We shouldnt need a constructor for a controller

Reply:
Im confused abt the constructors. On my local machine, removing the constructor from AppUserController the app runs fine, but if I remove it from a service class, the CommandLineRunner doesnt launch

Reply:
The constructors are necessary for autowiring fields. If u take a look at spring docs there are 3 ways to autowire: field level, method level and constructor level. The recommended way for a situation like this is constructor level. So All args constructor is necessary.

Reply:
The TAs in 321 didnt showed us field level but this is not best practice

Reply:
https://www.youtube.com/watch?v=aX-bgylmprA&ab_channel=DanVega